netmap/types.proto

```diff
@@ -130,6 +130,53 @@ message PlacementPolicy {
     json_name = "subnetId",
     deprecated = true
   ];
+
+  // Erasure coding rule for container objects.
+  //
+  // For each original object, the payload is split into `data_part_num` data
+  // and `parity_part_num` parity parts. Each part is the same size. Data parts
+  // contain the original payload. If its length is not divisible by
+  // `data_part_num`, the last part is aligned with zero bytes. Both
+  // `data_part_num` and `parity_part_num` MUST NOT be zero or exceed 64,
+  // including in total.
+  //
+  // For each payload part, a part object is created. Original object's ID,
+  // signature and header is written in `header.split.parent`,
+  // `header.split.parent_signature` and `header.split.parent_header` fields
+  // correspondingly. Part index is written in the `__NEOFS__EC_PART_IDX`
+  // attribute as base-10 integer. Rule index in `PlacementPolicy.ec_rules`
+  // list is written in the `__NEOFS__EC_RULE_IDX` attribute as base-10
+  // integer.
+  //
+  // Each part object is stored in the container in one copy. Storage nodes are
+  // selected from the network map similar to `PlacementPolicy.replicas` rules.
+  // Optional `selector` acts the same way. The object for the `i`-th part is
+  // placed in the `i`-th node. If it is unavailable, the backup nodes with
+  // indexes `m * n + i` (`n = data_part_num + parity_part_num`,
+  // `m = 1, ..., CBF-1`). If all nodes for the `i`-th part
+  // are unavailable, nodes for the `i+1`-th (0 for the last) part are tried,
+  // and so on.
+  //
+  // Once part objects are stored in the container, the original object remains
+  // available if at least `data_part_num` of any part objects are available.
+  // In other words, unavailability (including complete loss) of any of
+  // `parity_part_num` part objects does not violate availability of the
+  // original one.
+  //
+  // Objects of TOMBSTONE and LOCK types are not encoded and stored as they are
+  // because they have no payload.
+  message ECRule {
+    // Number of data parts
+    uint32 data_part_num = 1;
+
+    // Number of parity parts
+    uint32 parity_part_num = 2;
+
+    // Name of the linked selector
+    string selector = 3 [json_name = "selector"];
+  }
+  // Erasure coding rules. Limited to 256 items.
```

Comment:
Limit to 1? Yeah, I know it's an array and theoretically many things are possible. But I'm not sure what will happen with "EC 3/1 EC 12/4 EC 5/3 EC 4/2" policy. Maybe it'll work, but I doubt it'll be usable. Also it's easier to extend things in future rather than constrain. So maybe 2-4, but no more.

Reply:
i doubt various encoding parameters will be in demand too. But multiple settings may be required to bind to different selectors. For example, they can define disjoint subnets for placement

picked 256 from `replicas`

Reply:
Let's bring it down to four.